Code/src/main/java/nl/utwente/viskell/ghcj/GhciSession.java

```diff
@@ -37,7 +37,7 @@
     private Evaluator ghci;
 
     /** Gets filled with a HaskellCatalog instance when ghci is ready. */
-    private HaskellCatalog catalog;
+    private static HaskellCatalog catalog;
```

Comment:
I think it save to assume for now that only one active ghci session exist as you have only one active toplevel working area. Leave those bigger changes for when they are really needed.